src/snap.ts

```diff
@@ -1,20 +1,32 @@
 import {Wallet} from "./interfaces";
 import {getPublicKey} from "./rpc/getPublicKey";
 import {initApi} from "./polkadot/initApi";
-import {exportPrivateKey} from "./rpc/exportPrivateKey";
+import {exportSeed} from "./rpc/exportSeed";
+import {getBalance} from "./rpc/substrate/getBalance";
+import {getAddress} from "./rpc/getAddress";
+import ApiPromise from "@polkadot/api/promise";
 
 declare let wallet: Wallet;
 
 wallet.registerRpcMessageHandler(async (originString, requestObject) => {
+  // init api if substrate method called
+  let api: ApiPromise = null;
+  if (requestObject.method.startsWith("substrate")) {
+    api = await initApi();
+  }
   switch (requestObject.method) {
     case 'getPublicKey':
       return await getPublicKey(wallet);
+    case 'getAddress':
+      return await getAddress(wallet);
+    case 'exportSeed':
+      return await exportSeed(wallet);
+    case 'substrate_getBalance':
+      return await getBalance(wallet, api);
     case 'substrate_getChainHead':
-      const api = await initApi();
+      // temporary method
```

Comment:
why temporary?

Reply:
I used it just to test polkadot api, do we need to expose this ?